code/game/machinery/cryopod.dm

```diff
@@ -0,0 +1,430 @@
+				else
+					var/list/objectivestoupdate
+					for(var/datum/mind/own in O.get_owners())
+						to_chat(own.current, "<BR><span class='userdanger'>You get the feeling your target is no longer within reach. Time for Plan [pick("A","B","C","D","X","Y","Z")]. Objectives updated!</span>")
+						for(var/datum/objective/ob in own.get_all_objectives())
+							LAZYADD(objectivestoupdate, ob)
+					objectivestoupdate += O.team.objectives
+					for(var/datum/objective/ob in objectivestoupdate)
+						if(ob.target != old_target || !istype(ob,O.type))
+							return
+						ob.target = O.target
+						ob.update_explanation_text()
+						to_chat(O.owner.current, "<BR><span class='userdanger'>You get the feeling your target is no longer within reach. Time for Plan [pick("A","B","C","D","X","Y","Z")]. Objectives updated!</span>")
+						ob.owner.announce_objectives()
+				qdel(O)
+
+// This function can not be undone; do not call this unless you are sure
+/obj/machinery/cryopod/proc/despawn_occupant()
+	var/mob/living/mob_occupant = occupant
+
+	if(mob_occupant.mind && mob_occupant.mind.assigned_role)
+		//H andle job slot/tater cleanup.
+		var/job = mob_occupant.mind.assigned_role
+		SSjob.FreeRole(job)
+		if(LAZYLEN(mob_occupant.mind.objectives))
+			mob_occupant.mind.objectives.Cut()
+			mob_occupant.mind.special_role = null
+
+	// Delete them from datacore.
+	var/announce_rank = null
+	for(var/datum/data/record/R in GLOB.data_core.medical)
+		if((R.fields["name"] == mob_occupant.real_name))
+			qdel(R)
+	for(var/datum/data/record/T in GLOB.data_core.security)
+		if((T.fields["name"] == mob_occupant.real_name))
+			qdel(T)
+	for(var/datum/data/record/G in GLOB.data_core.general)
+		if((G.fields["name"] == mob_occupant.real_name))
+			announce_rank = G.fields["rank"]
+			qdel(G)
+
+	// Make an announcement and log the person entering storage.
+	if(control_computer)
+		control_computer.frozen_crew += "[mob_occupant.real_name]"
+
+	if(GLOB.announcement_systems.len)
+		var/obj/machinery/announcement_system/announcer = pick(GLOB.announcement_systems)
+		announcer.announce("CRYOSTORAGE", mob_occupant.real_name, announce_rank, list())
+		visible_message("<span class='notice'>\The [src] hums and hisses as it moves [mob_occupant.real_name] into storage.</span>")
+
+
+	for(var/obj/item/W in mob_occupant.GetAllContents())
+		if(W.loc.loc && (( W.loc.loc == loc ) || (W.loc.loc == control_computer)))
+			continue // means we already moved whatever this thing was in
+			// I'm a professional, okay
+		for(var/T in preserve_items)
+			if(istype(W, T))
+				if(control_computer && control_computer.allow_items)
+					control_computer.frozen_items += W
+					mob_occupant.transferItemToLoc(W, control_computer, TRUE)
+				else
+					mob_occupant.transferItemToLoc(W, loc, TRUE)
+
+	for(var/obj/item/W in mob_occupant.GetAllContents())
```

Comment:
QDEL_LIST

Reply:
would I still need a for loop in this case or could it simply be replaced with ```QDEL_LIST(mob_occupant.GetAllContents())```